extensions/email/src/main/java/org/mule/extension/email/api/AbstractEmailConnection.java

```diff
@@ -0,0 +1,264 @@
+     * @param username          the username to establish connection with the mail server.
+     * @param password          the password corresponding to the {@code username}
+     * @param host              the host name of the mail server.
+     * @param port              the port number of the mail server.
+     * @param connectionTimeout the socket connection timeout
+     * @param readTimeout       the socket read timeout
+     * @param writeTimeout      the socket write timeout
+     * @param properties        the custom properties added to configure the session.
+     * @param tlsContextFactory the tls context factory for creating the context to secure the connection
+     */
+    public AbstractEmailConnection(String protocol,
+                                   String username,
+                                   String password,
+                                   String host,
+                                   String port,
+                                   long connectionTimeout,
+                                   long readTimeout,
+                                   long writeTimeout,
+                                   Map<String, String> properties,
+                                   TlsContextFactory tlsContextFactory)
+    {
+        this.protocol = protocol;
+
+        Properties sessionProperties = buildSessionProperties(host, port, connectionTimeout, readTimeout, writeTimeout, tlsContextFactory);
+
+        if (properties != null)
+        {
+            sessionProperties.putAll(properties);
+        }
+
+        PasswordAuthenticator authenticator = null;
+        if (username != null && password != null)
+        {
+            set(sessionProperties, MAIL_AUTH_PROPERTY_MASK, "true");
+            authenticator = new PasswordAuthenticator(username, password);
+        }
+
+        session = Session.getInstance(sessionProperties, authenticator);
+    }
+
+
+    /**
+     * Creates a new instance and set all the properties required by the specified {@code protocol}.
+     */
+    private Properties buildSessionProperties(String host, String port, long connectionTimeout, long readTimeout, long writeTimeout, TlsContextFactory tlsContextFactory)
+    {
+        Properties properties = new Properties();
+        set(properties, PORT_PROPERTY_MASK, port);
+        set(properties, HOST_PROPERTY_MASK, host);
+
+        if (tlsContextFactory != null)
+        {
+            setSecureProperties(tlsContextFactory, properties);
+            // SSL
+            //local.setProperty("mail." + getProtocol() + ".ssl", "true");
+        }
+
+        set(properties, CONNECTION_TIMEOUT_PROPERTY_MASK, Long.toString(connectionTimeout <= 0L ? DEFAULT_TIMEOUT : connectionTimeout));
+        set(properties, TIMEOUT_PROPERTY_MASK, Long.toString(readTimeout <= 0L ? DEFAULT_TIMEOUT : readTimeout));
+
+        // Note: "mail." + protocol + ".writetimeout" breaks TLS/SSL Dummy Socket and makes tests run 6x slower!!!
+        if (writeTimeout > 0L)
```

Comment:
default Timeout should be used as default on the parameter and thus you should always assume that the value is set